### PR TITLE
cable coil cleanup

### PR DESCRIPTION
**Pull request by BatElite**
[cleanliness][QOL]
## About the PR 
This PR is a bit of scattershot of code cleanup regarding cable coils, largely because if you look at cable coil code long enough you get annoyed at all the bullshit it's reinventing. This PR also mirrors one I made on coolstation, 

-Making a new cable object is now its own little proc to save copy-paste.

-The four(!) procs for putting a cable on the floor have been merged into two (one targeting turfs and one other placed cables), which had considerable overlap anyway. Most of the output messages got axed in the process . I'm thinking if those have to stay around it's probably best to do something with return codes, since putting them directly in the procs can get spammy when autolaying.

-As a by-effect of the above, auto-laying cable can now also shock you if powered. The fact that this causes sparks from unpowered cables is a by-effect of the cable (object) shock proc being bugged, but that's not this PR's problem!

-Cable coils now use 'COMSIG_MOVABLE_MOVED' instead of 'move_callback' since the latter is more or less superceded by components anyway.

-Cable coils now use more of the default obj/item stack procs instead of reinventing them (specifically the take proc and 3/4rd of attackby). I left 'update_icon()' alive because I didn't want to change a bunch of calls to it in other files, but it could just be replaced with 'update_stack_appearance'

-Splitting a coil with wirecutters now lets you specify the length to cut instead of making 1 length coils, and also doesn't eat another unit of cable for the trouble. I don't know why either of those were ever thought to be a good idea.

-Removing and merging all that jazz brought down the file by about a third.

## Why's this needed? 
I think these changes are more or less all improvements (or at least trade-offs to lessen duplicate code) to an item sorely in need of a cleanup, but I can't really say I know what the hell the previous author(s) were thinking with some of this.

I guess splitting coils arbitrarily is kinda cool though.

### CODE REVIEW

File: code/obj/item/cable_coil.dm [hunk 1]
```diff
@@ -234,182 +225,65 @@ obj/item/cable_coil/abilities = list(/obj/ability_button/cable_toggle)
 
 /obj/item/cable_coil/attackby(obj/item/W, mob/user)
 	if (issnippingtool(W) && src.amount > 1)
-		src.amount--
-		take(1, user.loc)
-		boutput(user, "You cut a piece off the [base_name].")
-		src.UpdateIcon()
+		var/obj/item/cable_coil/A = split_stack(round(input("How long of a wire do you wish to cut?","Length of [src.amount]",1) as num))
+		if (istype(A))
+			A.set_loc(user.loc) //Hey, split_stack, Why is the default location for the new item src.loc which is *very likely* to be a damn mob?
```
Comment (ZeWaka) [inline code/obj/item/cable_coil.dm:230]: wouldn't we want get turf?

Comment (BatElite) [reply code/obj/item/cable_coil.dm]: I don't think it matters that much, but sure. It's a little weird if you were to cut cable while in a locker or something but frankly I doubt things like that occur in practice

File: code/obj/item/cable_coil.dm [hunk 2]
```diff
@@ -234,182 +225,65 @@ obj/item/cable_coil/abilities = list(/obj/ability_button/cable_toggle)
 
 /obj/item/cable_coil/attackby(obj/item/W, mob/user)
 	if (issnippingtool(W) && src.amount > 1)
-		src.amount--
-		take(1, user.loc)
-		boutput(user, "You cut a piece off the [base_name].")
-		src.UpdateIcon()
+		var/obj/item/cable_coil/A = split_stack(round(input("How long of a wire do you wish to cut?","Length of [src.amount]",1) as num))
+		if (istype(A))
+			A.set_loc(get_turf(user)) //Hey, split_stack, Why is the default location for the new item src.loc which is *very likely* to be a damn mob?
```
Comment (ZeWaka) [inline code/obj/item/cable_coil.dm:230]:
```suggestion
			A.set_loc(user.loc) //Hey, split_stack, Why is the default location for the new item src.loc which is *very likely* to be a damn mob?
```
ok. makes sense

**Review by ZeWaka — APPROVED** [review]

**Comment by UrsulaMejor** [comment]
This PR broke laying cable by click-dragging. Please fix.

**Comment by UrsulaMejor** [comment]
`/turf/simulated/floor/MouseDrop_T(atom/A, mob/user as mob)
	..(A,user)
	if(istype(A,/turf/simulated/floor))
		var/turf/simulated/floor/F = A
		var/obj/item/I = user.equipped()
		if(I)
			if(istype(I,/obj/item/cable_coil))
				var/obj/item/cable_coil/C = I
				if(BOUNDS_DIST(user,F) == 0 && BOUNDS_DIST(user,src) == 0)
					C.move_callback(user, F, src)`

**Comment by UrsulaMejor** [comment]
Nevermind, figured it out and fixed it myself.

**Pull request merged**